### PR TITLE
Couple bug fixes

### PR DESCRIPTION
**Pull request by LifeIsPizza**
- Ritual of Balance is now exhausted when used by an opponent (Nexus, Remote Access) even if the condition isn't met

- Mobius Scroll is now archived in the owner's archives when used by an opponent (Nexus, Remote Access).

- Same as above for Hexpion (which can change controller)

### CODE REVIEW

File: server/game/cards/01-Core/RitualOfBalance.js [hunk 1]
```diff
@@ -6,9 +6,13 @@ class RitualOfBalance extends Card {
             condition: context => context.player.opponent && context.player.opponent.amber >= 6,
             gameAction: ability.actions.steal()
         });
+        this.action({
```
Comment (jeremylarner) [inline server/game/cards/01-Core/RitualOfBalance.js]: I don't think this is the correct way to fix this bug.  The problem is that the game won't allow the artifact to be used when the action condition isn't met, whereas it should allow the ability to be used, and just not permit the game action to resolve

File: test/server/cards/01-Core/MobiusScroll.spec.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-describe('Mobius Scroll', function() {
+fdescribe('Mobius Scroll', function() {
```
Comment (cryogen) [inline test/server/cards/01-Core/MobiusScroll.spec.js]: please remove the 'f' here as it means all test wont run

Comment (LifeIsPizza) [reply test/server/cards/01-Core/MobiusScroll.spec.js]: Sorry forgot to revert it back

File: server/game/cards/01-Core/RitualOfBalance.js (hunk 1 above)
Comment (LifeIsPizza) [reply server/game/cards/01-Core/RitualOfBalance.js]: I'm sure there's a more clever and cleaner way to do it... Not sure how to put a condition inside the gameAction though, as there are no other cards that work in a similar way.
On top of my head, stuff like Anomaly Exploiter with no damaged creatures should exhaust just fine as the condition is on the selection...

Comment (jeremylarner) [reply server/game/cards/01-Core/RitualOfBalance.js]: Hmmm, I'll take a look

**Comment by cryogen** [comment]
@LifeIsPizza is this PR still valid?

There's a failing test (which might be correct, but ideally it needs fixing :) )

**Comment by cryogen** [comment]
@jeremylarner Is this something you're still looking at?

**Pull request merged**